File: 박재현/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>
using namespace std;

int solution(vector<int> people, int limit)
{
    int answer = 0;

    sort(people.begin(), people.end()); // 50 50 70 80

    int left = 0;
    int right = people.size() - 1;

    while (left <= right)
    { // 겹치는 순간 멈춤
        if (people[left] + people[right] <= limit)
        {
            left++;
        }
        right--;
        answer++;
    }

    return answer;
}

File: 윤석주/day8/구명보트.py
def solution(people, limit):
    answer = 0
    people.sort(reverse=True)
    start = 0
    end = len(people)-1
    while start<=end:
        boat = 0
        while boat+people[start] <= limit and start != end:
            boat += people[start]
            start+=1
        while boat+people[end] <= limit:
            boat += people[end]
            end-=1
        answer+=1
    return answer